app/controllers/BaseController.scala

```diff
@@ -19,17 +19,11 @@ package controllers
 import config.AppConfig
 import play.api.i18n.Messages
 import play.api.mvc.Request
-import uk.gov.hmrc.play.frontend.controller.FrontendController
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import views.html.error_template
 
-trait BaseController extends FrontendController {
-  implicit val appConfig: AppConfig
+abstract class BaseController(implicit val appConfig: AppConfig) extends FrontendController {
```

Comment:
I may be completely wrong, but from the looks of things you can remove the implicits from `error_template.scala.html` as none of them look to be used. This would then let you remove the implicit appConfig here and in the various places you override the config.